source/MaterialXTest/MaterialXCore/Traversal.cpp

```diff
@@ -171,3 +173,28 @@ TEST_CASE("Traversal", "[traversal]")
     REQUIRE(!output->hasUpstreamCycle());
     REQUIRE(doc->validate());
 }
+
+TEST_CASE("InterGraph Tranversal", "[traversal]")
+{
+    mx::FileSearchPath searchPath;
+    const mx::FilePath currentPath = mx::FilePath::getCurrentPath();
+    searchPath.append(currentPath / mx::FilePath("libraries"));
+
+    mx::DocumentPtr doc = mx::createDocument();
+    mx::loadLibraries({ "stdlib", "pbrlib", "bxdf" }, searchPath, doc);
+
+    mx::FilePath testFile = mx::FilePath::getCurrentPath() / mx::FilePath("resources/Materials/TestSuite/stdlib/nodegraph_inputs/nodegraph_nodegraph.mtlx");
+    mx::readFromXmlFile(doc, testFile, searchPath);
+    REQUIRE(doc->validate());
+
+    for (mx::NodeGraphPtr graph : doc->getNodeGraphs())
+    {
+        for (mx::InputPtr interfaceInput : graph->getInputs())
+        {
+            if (interfaceInput && (!interfaceInput->getNodeName().empty() || !interfaceInput->getNodeGraphString().empty()))
```

Comment:
Nitpick: is the null check on `interfaceInput` needed? I don't think graph->getInputs() can return a vector where some of the elements are null.

Reply:
Probably not. Just firewall paranoia :). I can remove.